src/app/dashboard/index.tsx

```diff
@@ -1,15 +1,52 @@
-import React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import React, { useState } from 'react';
 import styled from 'styled-components';
+import { useQueryParams, StringParam, NumberParam } from 'use-query-params';
 
 import StockChart from './components/chart';
 import StockChartFiltersForm from './components/filters/form';
+import { minDateBoundary } from './components/filters/utils';
 import data from './sampleChartData';
 
 const DashboardPage: React.FC = () => {
+  const [query, setQuery] = useQueryParams({
+    date_from: StringParam,
+    date_to: StringParam,
+    aggregate: NumberParam,
+  });
+  setQuery;
+
+  const [startDate, setStartDate] = useState(
+    dayjs(query.date_from) ?? minDateBoundary
+  );
+  const [endDate, setEndDate] = useState(dayjs(query.date_to) ?? dayjs());
+  const [aggregate, setAggregate] = useState(query.aggregate ?? 1);
+
+  function getDataFromServer(
+    startDate: Dayjs,
+    endDate: Dayjs,
+    aggregate: number
+  ) {
+    return data.filter((record) => {
+      const recordDateAsDayjsDate = dayjs(record.date);
+      aggregate;
+      return (
+        recordDateAsDayjsDate >= startDate && recordDateAsDayjsDate <= endDate
+      );
+    });
+  }
+
   return (
     <Container>
-      <StockChartFiltersForm />
-      <StockChart data={data} />
+      <StockChartFiltersForm
+        startDate={startDate}
+        endDate={endDate}
+        aggregate={aggregate}
+        setStartDateHandle={setStartDate}
+        setEndDateHandle={setEndDate}
+        setAggregateHandle={setAggregate}
+      />
+      <StockChart data={getDataFromServer(startDate, endDate, aggregate)} />
```

Comment:
W `query` mamy `string` więc obecnie chcemy przekazać tam object `Dayjs`

Ogólnie to jak będziemy się łączyć z backendem to wszystko będzie do wywalenia (łącznie z funkcją `getDataFromServer`), ale na teraz to wystarczy żeby działało

```suggestion
      <StockChart data={getDataFromServer(dayjs(query.date_from),  dayjs(query.date_to), aggregate)} />
```